### PR TITLE
refactor: break out ecdh unwrap

### PR DESCRIPTION
**Pull request by hko-s**

### CODE REVIEW

File: src/crypto/ecdh.rs [hunk 1]
```diff
@@ -91,70 +89,87 @@ impl Decryptor for SecretKey {
...
+            fingerprint,
+        )
+    }
+}
 
-            // Padding length seems to exceed size of the padded message
-            if *pad as usize > len {
-                return Err(Error::UnpadError);
-            }
+pub fn ecdh_decrypt_unwrap_unpad(
```
Comment (dignifiedquire) [inline src/crypto/ecdh.rs:110]: no need for the `ecdh` prefix, the module is already called `ecdh`

Comment (dignifiedquire) [reply src/crypto/ecdh.rs:110]: also the`unwrap` is a bit unfortunate, as in the rust context this means panic on error

Comment (dignifiedquire) [reply src/crypto/ecdh.rs:110]: also if this becomes public it should have a doc comment describing what exactly it does

Comment (hko-s) [reply src/crypto/ecdh.rs]: Good points. I've made adjustments.

I'm torn about my change to use `KeyParams`: It makes the function parameters a little more expressive, but adds a pointless `oid.clone()`. What do you think?

Comment (dignifiedquire) [reply src/crypto/ecdh.rs]: more expressive sounds good

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**